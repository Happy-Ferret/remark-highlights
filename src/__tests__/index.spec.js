const remark = require(`remark`);
const plugin = require(`../index`);

const runTestWithPluginOptions = (code, options = {}) => {
  const markdownAST = remark.parse(code);
  plugin(options)(markdownAST);
  expect(markdownAST).toMatchSnapshot();
};

describe(`remark atom highlights plugin`, () => {
  it(`generates highlighted code`, () => {
    const code = `\`\`\`js
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`does not blast off when no lang is provided`, () => {
    const code = `\`\`\`
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight code when language name is provided`, () => {
    const code = `\`\`\`javascript
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight code when filename name is provided`, () => {
    const code = `\`\`\`test.js
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`inline scopePrefix`, () => {
    const code = `\`\`\`js{scopePrefix: "source--"}
// Atom Highlight
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight codes from scope when scope is provided; it would try to use js even if rs is provided`, () => {
    const code = `\`\`\`rs{scopeName: "source.js"}
// Atom Highlight
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight codes from addition langs when passed in inline config`, () => {
    const code = `\`\`\`rs{languagePackage: "language-rust"}
// Atom Highlight
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`loads grammars from inline config`, () => {
    const code = `\`\`\`rs{languagePackage: "language-rust"}
// Atom Highlight
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight codes from scope when scope is provided and languagePackage is provided`, () => {
    const code = `\`\`\`html{scopeName:"source.rust", languagePackage:"language-rust"}
fn main() {}
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`should highlight line numbers`, () => {
    const code = `\`\`\`js{highlightLines: (1-2, 5)}
module.exports = (options) => (ast) => {
  console.log("This");
  console.log("is");
  console.log("a");
  console.log("test");
};
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight when inline config is provided in single quote`, () => {
    const code = `\`\`\`js{scopePrefix: 'source--'}
// Atom Highlight
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`should replace pre class from inlineconfig`, () => {
    const code = `\`\`\`js{ preClass: { removeClass: true } }
// Atom Highlight with fileName
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`has codewraps when inline codewrapping is enabled`, () => {
    const code = `\`\`\`js{ codeWrap: true }
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`has codewraps and fileicons when enabled in inline config`, () => {
    const code = `\`\`\`js{ codeWrap: true, showFileIcon: true  }
// Atom Highlights
\`\`\``;
    runTestWithPluginOptions(code);
  });

  it(`highlight codes from addition langs from plugin config`, () => {
    const code = `\`\`\`rs
// Atom Highlights
\`\`\``;
    const options = { additionalLangs: [`language-rust`] };
    runTestWithPluginOptions(code, options);
  });

  it(`has codewraps when codewrapping is enabled`, () => {
    const code = `\`\`\`js
// Atom Highlights
\`\`\``;
    const options = { codeWrap: true };
    runTestWithPluginOptions(code, options);
  });

  it(`has codewraps with specific class when codewrapping is enabled`, () => {
    const code = `\`\`\`js
// Atom Highlights
\`\`\``;
    const options = { codeWrap: { className: "midnight" } };
    runTestWithPluginOptions(code, options);
  });

  it(`override scopePrefix from config when inline config is provided`, () => {
    const code = `\`\`\`js{scopePrefix: ""}
// Atom Highlight
\`\`\``;
    const options = { scopePrefix: `source--` };
    runTestWithPluginOptions(code, options);
  });

  it(`should show filename`, () => {
    const code = `\`\`\`js
// Atom Highlight with fileName
\`\`\``;
    const options = { showFileName: true };
    runTestWithPluginOptions(code, options);
  });

  it(`should show fileicon`, () => {
    const code = `\`\`\`js
// Atom Highlight with fileName
\`\`\``;
    const options = { showFileIcon: true };
    runTestWithPluginOptions(code, options);
  });

  it(`should replace pre class from config`, () => {
    const code = `\`\`\`js
// Atom Highlight with fileName
\`\`\``;
    const options = { preClass: { removeClass: true } };
    runTestWithPluginOptions(code, options);
  });

  it(`should set custom pre class from config`, () => {
    const code = `\`\`\`js
// Atom Highlight with fileName
\`\`\``;
    const options = { preClass: { className: "foo bar" } };
    runTestWithPluginOptions(code, options);
  });
});
