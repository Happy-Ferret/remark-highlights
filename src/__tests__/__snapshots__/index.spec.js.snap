// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark atom highlights plugin does not blast off when no lang is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 26,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"text plain null-grammar\\"><span>//&nbsp;Atom&nbsp;Highlights</span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 26,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin generates highlighted code 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 28,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 28,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin has codewraps and fileicons when enabled in inline config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{ codeWrap: true, showFileIcon: true  }",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 67,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<span class='highlighted-file-icon js-icon medium-yellow'></span><div class='highlight'><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre></div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 67,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin has codewraps when codewrapping is enabled 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 28,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<div class='highlight'><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre></div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 28,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin has codewraps when inline codewrapping is enabled 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{ codeWrap: true }",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 46,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<div class='highlight'><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre></div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 46,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin has codewraps with specific class when codewrapping is enabled 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 28,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<div class='midnight'><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre></div>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 28,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight code when filename name is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "test.js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 33,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 33,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight code when language name is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "javascript",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 36,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 36,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight codes from addition langs from plugin config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "rs",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 28,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source rust\\"><span class=\\"comment line double-slash rust\\"><span>//</span><span>&nbsp;Atom&nbsp;Highlights</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 28,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight codes from addition langs when passed in inline config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "rs{languagePackage: \\"language-rust\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 61,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source rust\\"><span class=\\"comment line double-slash rust\\"><span>//</span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 61,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight codes from scope when scope is provided and languagePackage is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "html{scopeName:\\"source.rust\\", languagePackage:\\"language-rust\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 82,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source rust\\"><span class=\\"keyword other fn rust\\"><span>fn</span></span><span>&nbsp;</span><span class=\\"entity name function rust\\"><span>main</span></span><span>()&nbsp;</span><span>{</span><span>}</span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 82,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight codes from scope when scope is provided; it would try to use js even if rs is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "rs{scopeName: \\"source.js\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 51,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 51,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin highlight when inline config is provided in single quote 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{scopePrefix: 'source--'}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 52,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source--source source--js\\"><span class=\\"source--comment source--line source--double-slash source--js\\"><span class=\\"source--punctuation source--definition source--comment source--js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 52,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin inline scopePrefix 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{scopePrefix: \\"source--\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 52,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source--source source--js\\"><span class=\\"source--comment source--line source--double-slash source--js\\"><span class=\\"source--punctuation source--definition source--comment source--js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 52,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin loads grammars from inline config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "rs{languagePackage: \\"language-rust\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 61,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source rust\\"><span class=\\"comment line double-slash rust\\"><span>//</span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 61,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin override scopePrefix from config when inline config is provided 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{scopePrefix: \\"\\"}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 44,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 44,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should highlight line numbers 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{highlightLines: (1-2, 5)}",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 8,
          "offset": 166,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"editor editor-colors\\"><div class=\\"line highlighted-line\\"><span class=\\"source js\\"><span class=\\"support variable js\\"><span>module</span></span><span class=\\"meta function arrow js\\"><span class=\\"meta delimiter method period js\\"><span>.</span></span><span class=\\"entity name function js\\"><span>exports</span></span><span>&nbsp;</span><span class=\\"keyword operator assignment js\\"><span>=</span></span><span>&nbsp;</span><span class=\\"meta parameters js\\"><span class=\\"punctuation definition parameters begin bracket round js\\"><span>(</span></span><span class=\\"variable parameter function js\\"><span>options</span></span><span class=\\"punctuation definition parameters end bracket round js\\"><span>)</span></span></span><span>&nbsp;</span><span class=\\"storage type function arrow js\\"><span>=&gt;</span></span></span><span>&nbsp;</span><span class=\\"meta function arrow js\\"><span class=\\"meta parameters js\\"><span class=\\"punctuation definition parameters begin bracket round js\\"><span>(</span></span><span class=\\"variable parameter function js\\"><span>ast</span></span><span class=\\"punctuation definition parameters end bracket round js\\"><span>)</span></span></span><span>&nbsp;</span><span class=\\"storage type function arrow js\\"><span>=&gt;</span></span></span><span>&nbsp;</span><span class=\\"punctuation definition function body begin bracket curly js\\"><span>{</span></span></span></div><div class=\\"line highlighted-line\\"><span class=\\"source js\\"><span>&nbsp;&nbsp;</span><span class=\\"entity name type object console js\\"><span>console</span></span><span class=\\"meta method-call js\\"><span class=\\"meta delimiter method period js\\"><span>.</span></span><span class=\\"support function console js\\"><span>log</span></span><span class=\\"meta arguments js\\"><span class=\\"punctuation definition arguments begin bracket round js\\"><span>(</span></span><span class=\\"string quoted double js\\"><span class=\\"punctuation definition string begin js\\"><span>&quot;</span></span><span>This</span><span class=\\"punctuation definition string end js\\"><span>&quot;</span></span></span><span class=\\"punctuation definition arguments end bracket round js\\"><span>)</span></span></span></span><span class=\\"punctuation terminator statement js\\"><span>;</span></span></span></div><div class=\\"line\\"><span class=\\"source js\\"><span>&nbsp;&nbsp;</span><span class=\\"entity name type object console js\\"><span>console</span></span><span class=\\"meta method-call js\\"><span class=\\"meta delimiter method period js\\"><span>.</span></span><span class=\\"support function console js\\"><span>log</span></span><span class=\\"meta arguments js\\"><span class=\\"punctuation definition arguments begin bracket round js\\"><span>(</span></span><span class=\\"string quoted double js\\"><span class=\\"punctuation definition string begin js\\"><span>&quot;</span></span><span>is</span><span class=\\"punctuation definition string end js\\"><span>&quot;</span></span></span><span class=\\"punctuation definition arguments end bracket round js\\"><span>)</span></span></span></span><span class=\\"punctuation terminator statement js\\"><span>;</span></span></span></div><div class=\\"line\\"><span class=\\"source js\\"><span>&nbsp;&nbsp;</span><span class=\\"entity name type object console js\\"><span>console</span></span><span class=\\"meta method-call js\\"><span class=\\"meta delimiter method period js\\"><span>.</span></span><span class=\\"support function console js\\"><span>log</span></span><span class=\\"meta arguments js\\"><span class=\\"punctuation definition arguments begin bracket round js\\"><span>(</span></span><span class=\\"string quoted double js\\"><span class=\\"punctuation definition string begin js\\"><span>&quot;</span></span><span>a</span><span class=\\"punctuation definition string end js\\"><span>&quot;</span></span></span><span class=\\"punctuation definition arguments end bracket round js\\"><span>)</span></span></span></span><span class=\\"punctuation terminator statement js\\"><span>;</span></span></span></div><div class=\\"line highlighted-line\\"><span class=\\"source js\\"><span>&nbsp;&nbsp;</span><span class=\\"entity name type object console js\\"><span>console</span></span><span class=\\"meta method-call js\\"><span class=\\"meta delimiter method period js\\"><span>.</span></span><span class=\\"support function console js\\"><span>log</span></span><span class=\\"meta arguments js\\"><span class=\\"punctuation definition arguments begin bracket round js\\"><span>(</span></span><span class=\\"string quoted double js\\"><span class=\\"punctuation definition string begin js\\"><span>&quot;</span></span><span>test</span><span class=\\"punctuation definition string end js\\"><span>&quot;</span></span></span><span class=\\"punctuation definition arguments end bracket round js\\"><span>)</span></span></span></span><span class=\\"punctuation terminator statement js\\"><span>;</span></span></span></div><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"punctuation definition function body end bracket curly js\\"><span>}</span></span><span class=\\"punctuation terminator statement js\\"><span>;</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 8,
      "offset": 166,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should replace pre class from config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 41,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight&nbsp;with&nbsp;fileName</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 41,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should replace pre class from inlineconfig 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js{ preClass: { removeClass: true } }",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 76,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight&nbsp;with&nbsp;fileName</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 76,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should set custom pre class from config 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 41,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<pre class=\\"foo bar\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight&nbsp;with&nbsp;fileName</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 41,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should show fileicon 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 41,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<span class='highlighted-file-icon js-icon medium-yellow'></span><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight&nbsp;with&nbsp;fileName</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 41,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark atom highlights plugin should show filename 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 41,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<span class='highlighted-file-name'>js</span><pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"source js\\"><span class=\\"comment line double-slash js\\"><span class=\\"punctuation definition comment js\\"><span>//</span></span><span>&nbsp;Atom&nbsp;Highlight&nbsp;with&nbsp;fileName</span></span></span></div></pre>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 41,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
